### ISSUE TITLE
Search Results Pagination requires meaningful narration

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13537]
**Describe the bug**
- Pagination requires meaningful labels.
- Previous/Next Arrows and page numbers need more narration context.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Go to Digital Cameras menu, select option "Shop all Digital Cameras" to get several result pages.
2. Open any Screen Reader (VO, JAWS, NVDA)
3. Tab to Pagination
4. Hear missing context.

**Expected behavior**
- Narration has more detail. eg. "Product search pagination, Page 1 selected, link _[tab to end]_ Next Page, link" 

**Screenshots**
![Screen Shot 2021-08-05 at 7 04 49 PM](https://user-images.githubusercontent.com/15113219/130504497-c1813f36-9d1c-4e2c-992f-5606003de9e8.png)

**Comment by BenjaminEmiliani** [comment]
At the moment, the design of the pagination is to make inactive the current page of the list of results. 
So the aria-label="Page 1, current' example can't be obtained unless the design of the pagination component is changed.

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

It says: First Page, Last Page.
As stated above, it doesn't say current page, as it's inactive. 
![paginationFirstPage](https://user-images.githubusercontent.com/45409515/144912660-410e2563-bb6c-48f6-b2bd-74622bb53d26.png)
![paginationGrid](https://user-images.githubusercontent.com/45409515/144912665-6b3cd788-686e-4696-8aef-2f5bb6c96b01.png)
![paginationLast](https://user-images.githubusercontent.com/45409515/144912669-a7fc0e0f-0e9c-4660-937d-ba9192209016.png)
![paginationPage3](https://user-images.githubusercontent.com/45409515/144912672-9a262669-4f12-4bd0-94a7-f8889fda0678.png)
